R/profvis.R

```diff
@@ -171,12 +171,13 @@ profvis <- function(expr = NULL,
     if (remove_on_exit) {
       on.exit(unlink(prof_output), add = TRUE)
     }
-    repeat {
-      # Work around https://github.com/r-lib/rlang/issues/1749
-      eval(substitute(delayedAssign("expr", expr_q, eval.env = env)))
 
+    # Use unique name so we can easily trim below
```

Comment:
Probably needs some comments about why we use `new_function()` here. This is hard to explain but here is an attempt:

The goal is for the quoted expression to be directly called by a function that we can detect in both raw and simplified stack traces. In the simplified case, it's important for the function to directly call the profiled code and not just force a promise containing it, since in that case the sentinel function gets simplified away.

Reply:
Also need to document the behaviour change that we now evaluate in a child of `env`.